source/core/src/main/com/csse3200/game/components/gamearea/GameAreaDisplay.java

```diff
@@ -1,44 +1,316 @@
+    Image[] towers3 = new Image[3]; // Create an array for two towers
+
+
+    for (int i = 0; i < 2; i++) {
+      // Use "building1" for the first tower and "building2" for the second tower
+      skin.add("default", new Label.LabelStyle(new BitmapFont(), Color.WHITE));
+      skin.add("building1", new Texture("images/towers/WallTower.png"));
+      // Load textures for building1 and building2
+      towers1[i] = new Image(skin, "building1");
+      towers1[i].setBounds(Gdx.graphics.getWidth() * 40f / 100f, Gdx.graphics.getHeight() * 80f / 100f, 100, 100);
+      stage.addActor(towers1[i]);
+
+      final int towerIndex1 = i; // Capture the index in a final variable for the listener
+      towerPositions1[towerIndex1] = new Vector2(towers1[towerIndex1].getX(), towers1[towerIndex1].getY());
+
+      towers1[i].addListener(new InputListener() {
+        private float startX, startY;
+
+        public boolean touchDown(InputEvent event, float x, float y, int pointer, int button) {
+          startX = towers1[towerIndex1].getX();
+          startY = towers1[towerIndex1].getY();
+          Gdx.app.log("GameAreaDisplay", "Touch Down on Tower " + towerIndex1);
+          towerDetailsDialog1.setVisible(true);
+          towerDetailsDialog1.show(stage);
+
+          return true; // Return true to indicate that the event was handled
+        }
+
+        public void touchDragged(InputEvent event, float x, float y, int pointer) {
+          float deltaX = towers1[towerIndex1].getX() - startX;
+          float deltaY = towers1[towerIndex1].getY() - startY;
+
+          // Calculate the distance moved
+          float distanceMoved = (float) Math.sqrt(deltaX * deltaX + deltaY * deltaY);
+
+          // Check if the tower has been moved for a significant distance
+          if (!towerMoved1[towerIndex1] && distanceMoved >= 100f) {
+            // Decrement the number of towers and set the flag for this tower
+            numTowers1--;
+
+            towerMoved1[towerIndex1] = true; // Set the flag to indicate tower movement
+          }
+
+          // Implement dragging logic here
+          towers1[towerIndex1].moveBy(x - towers1[towerIndex1].getWidth() / 2,
+                  y - towers1[towerIndex1].getHeight() / 2);
+
+          towerPositions1[towerIndex1] = new Vector2(towers1[towerIndex1].getX(), towers1[towerIndex1].getY());
+          towerDetailsDialog1.setPosition(
+                  towerPositions1[towerIndex1].x + towers1[towerIndex1].getWidth(),
+                  towerPositions1[towerIndex1].y);
+
+        }
+
+        public void touchUp(InputEvent event, float x, float y, int pointer, int button) {
+          towerDetailsDialog1.hide();
+        }
+      });
+    }
+    for (int i = 0; i < 2; i++) {
+      // Use "building1" for the first tower and "building2" for the second tower
+      skin.add("default", new Label.LabelStyle(new BitmapFont(), Color.WHITE));
+      skin.add("building2", new Texture("images/towers/turret_deployed.png"));
```

Comment:
this "turret_deployed.png" image is no longer being used for anything in game

Reply:
sorry for that, We worked in our remote rep, changing that now